components/Embed/Embed.tsx

```diff
@@ -0,0 +1,439 @@
+                        })}
+                      style={{
+                        // Initialize hidden to prevent content flash.
+                        visibility: 'hidden'
+                      }}
+                    >
+                      <FollowMenu
+                        className={clsx(
+                          styles.menuButton,
+                          styles.followRssButton
+                        )}
+                        onOpen={handleFollowButtonClick}
+                        onClose={handleFollowCloseClick}
+                        isOpen={followShown}
+                        portalId="embed-modals"
+                        followUrls={followUrls}
+                      />
+
+                      <PlayerShareMenu
+                        className={clsx(styles.menuButton, styles.shareButton)}
+                        onOpen={handleShareButtonClick}
+                        onClose={handleShareCloseClick}
+                        embedHtml={embedHtml}
+                        isOpen={shareShown}
+                        portalId="embed-modals"
+                        isPrivate={isPrivate}
+                      />
+
+                      <SupportMenu
+                        className={clsx(
+                          styles.menuButton,
+                          styles.supportButton
+                        )}
+                        onOpen={handleSupportButtonClick}
+                        onClose={handleSupportCloseClick}
+                        isOpen={supportShown}
+                        portalId="embed-modals"
+                        supportUrls={supportUrls}
+                      />
+                    </div>
+                  </div>
+
+                  <div className={styles.progress}>
+                    <PlayerProgress />
+                    <WebMonetized
+                      className={styles.monetized}
+                      onOpen={handleWebMonetizationButtonClick}
+                      onClose={handleWebMonetizationCloseClick}
+                      isOpen={webMonetizationShown}
+                      portalId="embed-modals"
+                      paymentPointer={paymentPointer}
+                    />
+                  </div>
+                </div>
+              </div>
+
+              {canShowPlaylist && (
+                <div
+                  className={styles.playlist}
+                  {...(modalShown && {
+                    inert: 'inert'
+                  })}
+                >
+                  <Playlist style={{ height: '100%' }} />
+                </div>
+              )}
+
+              {modalShown && (
+                <div
+                  className={styles.modals}
+                  id="embed-modals"
+                  {...(!modalShown && {
+                    inert: 'inert'
+                  })}
+                />
+              )}
+            </Player>
+          )}
+        </div>
+      </div>
+    </>
+  );
+};
```

Comment:
Are there any changes in here, or largely copied whole from `pages/embed.tsx`?  (It's a big diff)

Reply:
Yes. Basically just moved that markup and hooks into its own component that could be shared on the embed and preview pages.